### ISSUE TITLE
[db] Unexpected Error with Repeated QueryBuilder Usage and db_audit

### ISSUE THREAD

**Issue by pinghe** [agnesoft/agdb#1122]
I've encountered an issue while working with the QueryBuilder and db_audit functions in our project. Here's a detailed description of the problem:

When I first execute the following statement:
````
QueryBuilder.insert().nodes()
    .aliases(["root"])
    .query();
````

and then call db_audit, the result is returned as expected.

However, when I repeat the same two steps, db_audit returns an error message:
````
trailing characters at line 1 column xxxx
````

I believe this issue is not expected behavior and might be a bug related to the handling of repeated queries or a state issue within the db_audit function.

Steps to Reproduce:

Execute the QueryBuilder.insert() with the specified aliases and query.
Call db_audit to get the initial result.
Repeat steps 1 and 2.
Expected Result:
The db_audit function should return the same result as the first call without any errors.

Actual Result:
After repeating the steps, db_audit returns an error with the message trailing 
````
characters at line 1 column xxxx.
````

**Comment by agnesoft** [comment]
@pinghe Thanks for the report. This is now fixed. The issue was in use of `serde_json` to concatenate the audit json. The `serde_json` does not do any concatenation at all on its own when writing to a file writer so I had to change it to do the concatenation manually.

Unfortunately the existing audit files produced up to now will be unusable as they have incorrect json format. They can be rescued  manually I suppose by opening the file in any text editor and replacing `][` sequence with `,`.
